crates/sui-core/src/authority/authority_test_utils.rs

```diff
@@ -97,9 +99,33 @@ pub async fn execute_certificate_with_execution_error(
         state_acc.accumulate_cached_live_object_set_for_testing(include_wrapped_tombstone);
 
     if with_shared {
-        send_consensus(authority, &certificate).await;
-        if let Some(fullnode) = fullnode {
-            send_consensus(fullnode, &certificate).await;
+        if fake_consensus {
+            send_consensus(authority, &certificate).await;
+            if let Some(fullnode) = fullnode {
+                send_consensus(fullnode, &certificate).await;
+            }
+        } else {
+            // Just set object locks directly if send_consensus is not requested.
+            authority
+                .epoch_store_for_testing()
+                .assign_shared_object_versions_for_tests(
+                    authority.get_object_cache_reader().as_ref(),
+                    &vec![VerifiedExecutableTransaction::new_from_certificate(
+                        certificate.clone(),
+                    )],
+                )
+                .await?;
+            if let Some(fullnode) = fullnode {
```

Comment:
I think for the fullnode case, we should always use this code instead of sending through process consensus since there is never consensus on fullnodes

Reply:
done